src/App.tsx

```diff
@@ -1,24 +1,131 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { ErrorType, Filter } from './types/type';
 
-const USER_ID = 0;
+const USER_ID = 52;
 
 export const App: React.FC = () => {
+  const [posts, setPosts] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [error, setError] = useState<ErrorType>({
+    load: false,
+    titleEmpty: false,
+    addTodo: false,
+    deleteTodo: false,
+    updateTodo: false,
+  });
+
+  const hasAnyError = Object.values(error).some(value => value === true);
+
+  const resetError = () => {
+    setTimeout(() => {
+      setError((prevState) => ({
+        ...prevState,
+        load: false,
+        titleEmpty: false,
+        deleteTodo: false,
+        updateTodo: false,
+        addTodo: false,
+      }));
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todos) => setPosts(todos))
+      .catch(() => {
+        setError((prevState) => ({
+          ...prevState,
+          load: true,
+        }));
+        resetError();
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return posts.filter(todo => {
+      switch (filter) {
+        case 'active':
```

Comment:
Use enum for this word
<img width="292" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/64750958/7b6a8c3e-585b-4965-834b-a2e1b4b79225">